### PR TITLE
Use the service library

### PR DESCRIPTION
**Pull request by dmpotter44**
This ends up being a fairly major rewrite but it mostly moves most of the shared code into the `clinical-trial-matching-service` branch.

Note that this is currently a draft until the base service PR is merged, so that the `package.json` can be changed to reference the `master` branch and not the current PR branch.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,49 +1,47 @@
...
+3. The service will now be running at http://localhost:3000/
+
+# Testing
+
+A validation test is provided to validate the ResearchStudy created via this service. Put an example response object in `spec/data/trial_object.json` and this object will be loaded and validated within this test. For this test to run, you must:
+
+1. Have a Java runtime environment installed
+2. [Download a copy of the HL7 validator JAR] (https://storage.googleapis.com/ig-build/org.hl7.fhir.validator.jar) into `spec/data/org.hl7.fhir.validator.jar`
+3. Have properly filled out `spec/data/trial_object.json`
+4. Have implemented the conversion function in `src/researchstudy-mapping.ts`
```
Comment (zlister) [inline README.md]: This should be updated with the new use of validation.

File: spec/validator.spec.ts [hunk 2]
```diff
@@ -16,7 +17,7 @@ import { ResearchStudy } from '../src/research-study';
  * being made properly.
  */
 import { exec } from 'child_process';
-import * as fs from 'fs';
+import fs from 'fs';
```
Comment (zlister) [inline spec/validator.spec.ts]: This file should probably be updated with the new validation work, once it is ready.

Comment (dmpotter44) [reply spec/validator.spec.ts]: This has been done.

File: README.md (hunk 1 above)
Comment (dmpotter44) [reply README.md]: The README has been updated with the new validator.

**Review by zlister — APPROVED** [review]

File: src/query.ts [hunk 3]
```diff
@@ -1,139 +1,262 @@
...
+  for (const trial of response.matchingTrials) {
+    if (isQueryTrial(trial)) {
+      studies.push(convertToResearchStudy(trial, id++));
+    } else {
+      // This trial could not be understood. It can be ignored if that should
+      // happen or raised/logged as an error.
+    }
+  }
+  // Generate the final SearchSet
+  return new SearchSet(studies);
```
Comment (lelevine) [inline src/query.ts:217]: This is a change that will need to happen in the service library, but the SearchSet constructor should be able to take in a list of match scores that correspond to the list of studies. That change will then require an update in this function to account for the addition of scores.

**Review by lelevine — APPROVED** [review]
Added separate searchset ticket to the backlog.

**Pull request merged**